### PR TITLE
chore(build): remove build process from repo

### PR DESCRIPTION
**Pull request by mittachaitu**
Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

## Pull Request template

Please, go through these steps before you submit a PR. ## Remove this line

**Why is this PR required? What issue does it fix?**:
This PR is required to incorporate CDDL license comments on cStor repo.

**What this PR does?**:
This PR does the following changes:
- Moves the main process i.e zrepl from openebs/cstor to openebs/libcstor.
- Move the build process from openebs/cstor to openebs/libcstor.
- Updates the downstream repo from openebs/istgt to openebs/libcstor.
- Updates the fio version in Travis to fix flakey issues.

**Does this PR require any upgrade changes?**:
No

**If the changes in this PR are manually verified, list down the scenarios covered and commands you used for testing with logs:**

**Any additional information for your reviewer?**:
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

**Review by mynktl — DISMISSED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -169,7 +164,7 @@ script:
 
         REL_BRANCH=$(echo $(echo "$TRAVIS_TAG" | cut -d'-' -f1 | rev | cut -d'.' -f2- | rev).x$REL_SUFFIX);
 
-        ./buildscripts/git-release "$REPO_ORG/istgt" "$TRAVIS_TAG" "$REL_BRANCH" || travis_terminate 1;
+        ./buildscripts/git-release "$REPO_ORG/libcstor" "$TRAVIS_TAG" "$REL_BRANCH" || travis_terminate 1;
```
Comment (vishnuitta) [inline .travis.yml:164]: what is this steps supposed to do? why this change required?

File: .travis.yml [hunk 2]
```diff
@@ -110,6 +96,17 @@ install:
         make --no-print-directory -s pkg-utils pkg-kmod || travis_terminate 1;
         sudo dpkg -i *.deb || travis_terminate 1;
       fi
+    # Return to libcstor code to complie zrepl which is responsible for uZFS feature.
+    # Save the current location to get back
+    - pushd .
+    - cd ..
```
Comment (vishnuitta) [inline .travis.yml]: we can just do with one 'cd' command, and after doing 'make', popd can be performed. No need to do 'cd ../..'

File: tests/cstor/gtest/gtest_utils.cc [hunk 3]
```diff
@@ -64,7 +64,14 @@ std::string GtestUtils::getCmdPath(std::string zfsCmd) {
 	std::string cmdPath;
 	const char *srcPath = std::getenv("SRC_PATH");
 
-	if (srcPath == nullptr) {
+	// Since the zrepl has been moved from cstor
+	// to libcstor base path of zrepl will be ../libcstor
```
Comment (pawanpraka1) [inline tests/cstor/gtest/gtest_utils.cc:68]: we should probably move the test also to the libcstor.

File: .travis.yml [hunk 4]
```diff
@@ -110,6 +96,17 @@ install:
...
+    - cd libcstor
+    # Go to zrepl directory to build zrepl
+    - cd cmd/zrepl
+    - make
+    - cd ../../
```
Comment (pawanpraka1) [inline .travis.yml]: as vitta said, no need to do cd ../../ just popd is sufficient.

File: tests/cstor/gtest/gtest_utils.cc (hunk 3 above)
Comment (mittachaitu) [reply tests/cstor/gtest/gtest_utils.cc:68]: Can I do it in separate PR?

File: .travis.yml (hunk 1 above)
Comment (mittachaitu) [reply .travis.yml:164]: This will help to trigger the release on the downstream repo from Travis... Since with this PR, images will be built from libcstor after i.e only after completing the libcstor Travis images & code will be pushed to corresponding image repositories and source repositories... 

Existing system: Since images are pushed from cstor triggering istgt release from cstor is good(With this PR things will change).
 
Release flow(For creating release images):
Existing flow: openebs/linux-utils ---> openebs/libcstor ---> openebs/cstor ---> openebs/istgt ....
Updated flow: openebs/linux-utils ---> openebs/cstor ---> openebs/libcstor ---> openebs/istgt ....

File: tests/cstor/gtest/gtest_utils.cc (hunk 3 above)
Comment (pawanpraka1) [reply tests/cstor/gtest/gtest_utils.cc:68]: surr @sai.

Comment (mittachaitu) [reply tests/cstor/gtest/gtest_utils.cc:68]: Thanks

File: .travis.yml (hunk 2 above)
Comment (mittachaitu) [reply .travis.yml]: Updated PTAL

File: .travis.yml (hunk 4 above)
Comment (mittachaitu) [reply .travis.yml]: Updated PTAL

File: .travis.yml (hunk 1 above)
Comment (pawanpraka1) [reply .travis.yml:164]: @mittachaitu let us keep the old flow, as libcstor will become the main repo later on (we will probably rename it) which will tag all dependant repo.

**Review by pawanpraka1 — COMMENTED** [review]
looks good. few minor comments.

**Comment by pawanpraka1** [comment]
@mittachaitu add the changelog in this PR.

File: .travis.yml (hunk 1 above)
Comment (mittachaitu) [reply .travis.yml:164]: Ok so you mean cstor and istgt will be triggered from libcstor itself(after renaming it).... that makes sense.

Then I no need to raise a PR in openebs/linux-utils cc: @kmova

Comment (pawanpraka1) [reply .travis.yml:164]: I see, we need base image first. We can build everything from libcstor. and cstor will be just a dependency.

Comment (mittachaitu) [reply .travis.yml:164]: We are building a base image from libcstor itself https://github.com/openebs/libcstor/blob/35c939ccb7473e2674a3fe9877c33cd74115740a/build_image.sh#L120. So can I change it to(Since base and main images is built from libcstor)
```
                                                   --------> openebs/cstor
openebs/linux-utils ---> openebs/libcstor  ----->  |
                                                   --------> openebs/istgt ----> other dependent repos
```

Comment (pawanpraka1) [reply .travis.yml:164]: @mittachaitu I see a problem. The image should be pushed at last. And since we are now pushing from libcstor, that should be done after build is successful on cstor. So cstor should be tagged first. 

The other thing is, we any way build everything from libcstor, so tagging cstor does not serve any purpose. just that we have marked the code for a particular tag.

let us go ahead with the way you did, later on we will decide how can we do that.

Comment (mittachaitu) [reply .travis.yml:164]: Can I remove the triggering downstream repo from cstor?

Comment (mittachaitu) [reply .travis.yml:164]: > let us go ahead with the way you did, later on we will decide how can we do that.

Ok

Comment (vishnuitta) [reply .travis.yml:164]: yeah.. triggering builds from cstor can be removed.

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Review by mynktl — APPROVED** [review]

**Comment by vishnuitta** [comment]
this PR and libcstor PR https://github.com/openebs/libcstor/pull/70/ are dependent on each other.. so, merging this PR even if the travis is failing
